### PR TITLE
Implemented DropdownSearch component for county and city. Closes #147.

### LINKED ISSUE #147: Implement county/city component
As part of the Stam Acasa project, we realized we need to develop 2 components for inserting information about County/City in Romania.

Screenshot 2020-04-09 at 12 17 33

The components should use JSON files for configuration.
The components should be able to return the data entered by an user.
The components are dropdown or searchable - if I use the dropdown function I can see all the elements, if I start typing the component does a search through the possible values. The city component depends on the county one, we should not be able to interact with the city component until the county one is filled in with a value.

### PR DESCRIPTION
**Pull request by dragosrobertn**
### What changed?
- Implemented a new component for Dropdown Search to be used for city and county as displayed in #147.
- This closes #147.
- Version was bumped to 1.0.11 and component exported at top level in `index.js`
image

### CODE REVIEW

File: src/components/dropdown-search/dropdown-search.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  values: [],
+  onSelect: () => {},
+  isAlwaysOpen: false,
+  showSearchInput: true,
+  searchPlaceholder: ""
+};
+
+DropdownSearch.propTypes = {
+  title: PropTypes.string.isRequired,
+  values: PropTypes.array.isRequired,
```
Comment (cipick) [inline src/components/dropdown-search/dropdown-search.js:101]: I would rename this prop `options` and would expect 
```
PropTypes.arrayOf(
  PropTypes.shape({
    value:PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]),
    label: PropTypes.string
  })
)
```
as the values from the database could be different than what user should see. What do you think?

Comment (cipick) [reply src/components/dropdown-search/dropdown-search.js:101]: this is how the model on backend will look like: https://github.com/code4romania/stam-acasa/pull/169/files#diff-10e2322e3b329f8be005addfd05e4368, we want to keep the ids in `value` key and the names in `label` key

File: src/components/dropdown-search/dropdown-search.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  onSelect: PropTypes.func,
```
Comment (cipick) [inline src/components/dropdown-search/dropdown-search.js:102]: i think this should be required too

File: src/components/dropdown-search/dropdown-search.js (hunk 1 above)
Comment (dragosrobertn) [reply src/components/dropdown-search/dropdown-search.js:101]: That's cool, and just to clarify, the returned value to onSelect should be the ID, and not the label, right? Or do we want both?

File: src/components/dropdown-search/dropdown-search.js (hunk 2 above)
Comment (dragosrobertn) [reply src/components/dropdown-search/dropdown-search.js:102]: That's ok I will be changing it :)

File: src/components/dropdown-search/dropdown-search.js (hunk 1 above)
Comment (cipick) [reply src/components/dropdown-search/dropdown-search.js:101]: `selectDropdownOption(option.value)` should be enough

File: src/components/dropdown-search/dropdown-search.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  return (
+    <div>
+      <div className="dropdown-search-filter" onClick={toggleDropdownOptions}>
+        <div className="dropdown-search-filter__name">{dropdownTitle}</div>
+        <span className="icon is-small is-right dropdown-search-filter__icon">
+          <ArrowDown />
+        </span>
+      </div>
+      {(showDropdownOptions || isAlwaysOpen) && (
+        <div className={"dropdown-search-options"}>
```
Comment (cipick) [inline src/components/dropdown-search/dropdown-search.js:55]:
```suggestion
        <div className="dropdown-search-options">
```

File: src/components/dropdown-search/dropdown-search.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import "./dropdown-search.scss";
+import ArrowDown from "../../images/icons/arrow-down.svg";
+import { SearchInput } from "../search-input/search-input";
+
+export const DropdownSearch = ({
+  title,
+  options,
+  onSelect,
+  isAlwaysOpen,
+  showSearchInput = true,
```
Comment (cipick) [inline src/components/dropdown-search/dropdown-search.js:12]: It gets the default value from `defaultProps`, please remove it from here

File: src/components/dropdown-search/dropdown-search.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+              })}
+            </div>
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+DropdownSearch.defaults = {
```
Comment (cipick) [inline src/components/dropdown-search/dropdown-search.js:90]:
```suggestion
DropdownSearch.defaultProps. = {
```

File: src/components/dropdown-search/dropdown-search.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  title: "",
+  values: [],
```
Comment (cipick) [inline src/components/dropdown-search/dropdown-search.js:92]: This is not used anymore

File: src/components/dropdown-search/dropdown-search.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  title,
+  options,
+  onSelect,
+  isAlwaysOpen,
+  showSearchInput = true,
+  searchPlaceholder
+}) => {
+  const [showDropdownOptions, setShowDropdownOptions] = useState(false);
+  const [dropdownTitle, setDropdownTitle] = useState(title);
+  const [dropdownOptions, setDropDownOptions] = useState(options);
```
Comment (cipick) [inline src/components/dropdown-search/dropdown-search.js:17]: Would be nice to call this method `setDropdownOptions` as we don't have `DropDown` in other places 😁

Comment (dragosrobertn) [reply src/components/dropdown-search/dropdown-search.js:17]: Done.

File: src/components/dropdown-search/dropdown-search.js (hunk 6 above)
Comment (dragosrobertn) [reply src/components/dropdown-search/dropdown-search.js:92]: It's `options` now so I will change it to that.

File: src/components/dropdown-search/dropdown-search.js (hunk 5 above)
Comment (dragosrobertn) [reply src/components/dropdown-search/dropdown-search.js:90]: Done.

File: src/components/dropdown-search/dropdown-search.js (hunk 4 above)
Comment (dragosrobertn) [reply src/components/dropdown-search/dropdown-search.js:12]: Done.

File: src/components/dropdown-search/dropdown-search.js (hunk 3 above)
Comment (dragosrobertn) [reply src/components/dropdown-search/dropdown-search.js:55]: Done.

**Pull request merged**